lib/engines/hcktest/playlist.rb

```diff
@@ -69,19 +93,20 @@ def custom_playlist(log)
     end
 
     def custom_ignore_list(log)
-      ignore_list = @project.engine.target['ignore_list']
-      @ignored_list = []
-      if ignore_list
-        @tests.reject! do |test|
-          if ignore_list.include?(test['name'])
-            @ignored_list << test
-            true
-          else
-            false
-          end
-        end
-      end
-      @logger.info('Applying custom ignore list') if log && ignore_list
+      custom_ignore_list_file = @project.options.test.ignore_list
+
+      ignore_list = if custom_ignore_list_file.nil?
+                      @project.engine.target['ignore_list']
+                    else
+                      File.readlines(custom_ignore_list_file).map(&:chomp)
+                    end
+
+      return unless ignore_list
+
+      @ignored_list = @tests.filter { ignore_list.include?(_1['name']) }
```

Comment:
I prefer `reject` rather than `filter` as the old code did. `filter` is somewhat unclear if it's selecting or rejecting.

Reply:
In the case of old code with `reject` after adding a condition for `File.readlines` we have a big `rubocop metrics/abcsize`. That's why I rewrite the mechanism. 

`@ignored_list = @tests.filter { ignore_list.include?(_1['name'])` - select ignored test 
`@tests -= @ignored_list` - reject ignored tests from list

Reply:
simplified old source and back logic to use `reject`

Reply:
I didn't meant to use `reject!` but to use `reject`, which is an alias of `filter`. Either of `reject` and `reject!` is fine in my opinion.